wallet/wallet-controller/src/lib.rs

```diff
@@ -132,6 +132,41 @@ impl<T: NodeInterface + Clone + Send + Sync + 'static> Controller<T> {
         Ok(wallet)
     }
 
+    /// Encrypts the wallet using the specified `password`, or removes the existing encryption if `password` is `None`.
+    ///
+    /// # Arguments
+    ///
+    /// * `password` - An optional `String` representing the new password for encrypting the wallet.
+    ///
+    /// # Returns
+    ///
+    /// This method returns an error if the wallet is locked
+    pub fn encrypt_wallet(&mut self, password: &Option<String>) -> Result<(), ControllerError<T>> {
+        self.wallet.encrypt_wallet(password).map_err(ControllerError::WalletError)
+    }
+
+    /// Unlocks the wallet using the specified password.
+    ///
+    /// # Arguments
+    ///
+    /// * `password` - A `String` representing the password that was used to encrypt the wallet.
+    ///
+    /// # Returns
+    ///
+    /// This method returns an error if the password is incorrect
+    pub fn unlock_wallet(&mut self, password: &String) -> Result<(), ControllerError<T>> {
+        self.wallet.unlock_wallet(password).map_err(ControllerError::WalletError)
+    }
+
```

Comment:
There's a typo "succesfully" in this file. Can you please fix it?

Reply:
fixed